sympy/combinatorics/fp_groups.py

```diff
@@ -545,24 +545,34 @@ def compute_polycyclic_series(self):
         return pc_series
 
 
-def subgroup_quotient(G, H):
+def subgroup_quotient(free_group, fp_grp, G, H):
     '''
     Compute the quotient group G/H
     where, H is the subgroup of G.
     '''
-    if not (isinstance(G, FpGroup)
-                and isinstance(H, FpGroup)):
-        raise ValueError("The group must be an instance of FpGroup")
-
-    if not G.free_group == H.free_group:
-        raise ValueError("The elements must belong"
-                                "to the same FreeGroup")
-
-    if not H.is_subgroup(G):
-        raise ValueError("H must be a subgroup of G")
+    if not isinstance(fp_group, FpGroup):
+        raise ValueError("The parent group must be an instance"
+                                "of FpGroup")
+
+    if isinstance(G, list):
+        g_gens, g_rels, _gens = reidemeister_presentation(fp_group, G, rel=True)
+        _homomorphism = homomorphism(G, fp_group, g_gens, _gens, check=False)
+        g_gens = _gens
+        g_rels = _homomorphism(g_rels)
+    else:
+        g_gens = G.generators
+        g_rels = G.relators
+
+    if isinstance(H, list):
+        h_gens, h_rels, _gens = reidemeister_presentation(fp_group, H, rel=True)
+        _homomorphism = homomorphism(H, fp_group, h_gens, _gens, check=False)
+        h_gens = _gens
+        h_rels = _homomorphism(h_rels)
+    else:
+        h_gens = H.generators
+        h_rels = H.relators
```

Comment:
These two `if` statements are almost identical - they can be a subroutine that you first call on `H` and then on `G`

Reply:
Still relevant